src/Events/SEO/Provider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Manages the legacy view removal and messaging.
+ *
+ * @since   5.13.0
+ *
+ * @package TEC\Events\Legacy\Views\V1
+ */
+
+ namespace TEC\Events\SEO;
+
+use TEC\Common\Contracts\Service_Provider;
+use TEC\Events\SEO\No_Index;
+
+
+/**
+ * Class Provider
+ *
+ * @since   5.13.0
+
+ * @package TEC\Events\Legacy\Views\V1
+ */
+class Provider extends Service_Provider {
+	public function register() {
+		$this->container->singleton( static::class, $this );
+
+		$this->hooks();
+	}
+
+	public function hooks() {
```

Comment:
No need to break out the hooking of one action into a dedicated hooking method, just leave it in `register`.
While I'm there, I suggest to make this a Controller and integrate the functions currently in the `No_Index` class in here.